gitbook-docs/market/economics.md

```diff
@@ -16,109 +18,164 @@ Proof Market constants:
+
+* Proof generation time.
+  Proof producers provide time bounds _t\_p_ in which they can generate proof for the statement.
+  Users can set the desired proof generation time _t\_r_.
+  Only proposals with the _t\_p_ ≤ _t\_r_ can be matched with the user’s request.
+* Proof producers rating.
+  Proof producer's rating depends on the rate of completed orders — orders where the proof producer
+  has provided correct proof in the requested time interval.
+
+Here are the basic rules of order matching for a request with specified time _t\_r_ and price _c\_r_:
+
+1. Choose proposals with _t\_p_ ≤ _t\_r_ and _c\_p_ ≤ _c\_r_.
+2. Sort them by price and proof generation time.
+   The user’s request defines the priority of these parameters.
 3. Apply proof producer's rating to the sorted list.
-4. Choose the top ask from the list.
+4. Choose the top proposal from the list.
 
-### Bids & Asks resale
+### Requests & proposals resale
 
-When creating a new bid, there may be situations when there is already an equivalent bid in the database. Two bids are considered equivalent if they match in the **statement** and **public\_input** fields. Therefore, when adding a new bid, the application first tries to find an equivalent bid in the system. If such a bid is found and there is already proof for it, the following happens:
+There may be situations when an equivalent request is already in the database.
+Two requests are considered equivalent if they match in the **statement** and **public\_input** fields.
+Upon receiving a new request, the application tries to find an equivalent request in the system.
+This is what happens if such a request is found and there's already proof for it:
 
-* the new bid immediately goes into the **completed** state;
-* the proof requester pays a commission to the proof market;
-* the proof requester pays a commission to the proof producer who generated the proof for the existing bid;
-* the proof requester pays a commission to the owner of the existing bid;
-* unspent tokens are returned to the proof requester.
+* The new request immediately goes into the **completed** state.
+* The proof requester pays commissions to the Proof Market, the proof producer who generated the proof
+  for the existing request, and the proof requester of the original request.
+* Unspent tokens are returned to the proof requester.
 
-It is assumed that the total amount of all commissions will be significantly lower than the market price of the bid. This approach allows a proof-requester to recoup its bid costs in the future (if he is the owner of the bid whose result will be reused) and even earn. Similarly, a proof producer can sell his proof more than once, and the proof producer will receive his commission from each such additional sale.
+The total amount of all commissions should be significantly lower than the market price of the request.
+This approach allows the request owner, whose result will be reused, to recoup request costs
```

Comment:
We've got some ambiguity in the terms here.

There are two types of requests, so a request owner can be both a proof requester and a proof producer. Let's avoid using the term "request owner". Instead, use "proof requester" and "proof producer".

```suggestion
This approach allows the proof requester, whose proof will be reused, to recoup proof costs
```